renderdoc/driver/vulkan/vk_initstate.cpp

```diff
@@ -577,25 +582,22 @@ bool WrappedVulkan::Prepare_InitialState(WrappedVkRes *res)
   else if(type == eResAccelerationStructureKHR)
   {
     VkResourceRecord *record = GetResourceManager()->GetResourceRecord(id);
-
     if(!record->accelerationStructureInfo->accelerationStructureBuilt)
     {
       RDCDEBUG("Skipping AS %s as it has not been built", ToStr(id).c_str());
       return true;
     }
 
-    VulkanAccelerationStructureManager::ASMemory result;
-    VkAccelerationStructureKHR as = ToUnwrappedHandle<VkAccelerationStructureKHR>(res);
-    if(!GetAccelerationStructureManager()->Prepare(as, m_QueueFamilyIndices, result))
-    {
-      SET_ERROR_RESULT(m_LastCaptureError, ResultCode::OutOfMemory,
-                       "Couldn't allocate readback memory");
-      m_CaptureFailure = true;
-      return false;
-    }
+    // Skip empty AS input data (BLASes are force ref-ed)
+    if(record->accelerationStructureInfo->memSize == 0)
+      return true;
```

Comment:
Is an empty AS invalid, or why do we want to skip that here? I'm assuming you're meaning with the comment that we might be serialising initial states for BLASs that haven't been built.

If we skip serialising here we're not going to rebuild it (unless we add handling for `Create_InitialState`) but I think both an empty BLAS and empty TLAS are degenerate but valid so if you're wanting to check if the AS has been built or not that should be checked for specifically I think, not based on the size of the data.

Reply:
I'm not sure what I was thinking about when I wrote that comment, as the line actually relates to this one: https://github.com/baldurk/renderdoc/pull/3451/files/e2342e8fa987befcb18292e5c0a2be5088c28643#diff-0fc27a78a09357936e822ace8f29fee4f432a61c7a42113f780104ccbf07911cR539.

I had seen some TLASs that had zero `VkAccelerationStructureBuildGeometryInfoKHR::geometryCount` in a tech demo, so the VVL would kick off about zero sized buffer allocations, copies, etc.  so I added an early exit in `CopyInputBuffers(..)` and ths corresponding one during the prepare stage.  Re-reading the spec, it expressly states that the `geometryCount` must be 1 for a TLAS, but says nothing for a BLAS so I guess it is valid.  I'll update the code so that it skips trying to do anything with empty buffers but otherwise processes it as normal.